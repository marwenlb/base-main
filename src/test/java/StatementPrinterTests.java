import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.*;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);

        verify(result);
    }
        @Test
    void testHTMLInvoice() {
        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        HTMLStatementPrinter statementPrinter = new HTMLStatementPrinter();
        String result = statementPrinter.print(invoice, plays);

        verifyHtml(result);
        
        // Vous pouvez également ajouter des assertions pour vérifier le contenu du rendu HTML si nécessaire.
    }

    @Test
    void statementWithNewPlayTypes() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("henry-v",  new Play("Henry V", "history"));
        plays.put("as-like",  new Play("As You Like It", "pastoral"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("henry-v", 53),
                new Performance("as-like", 55)));

        StatementPrinter statementPrinter = new StatementPrinter();
        Assertions.assertThrows(Error.class, () -> {
            statementPrinter.print(invoice, plays);
        });
    }
    
  @Test
    void testAudienceForTragedy() {
    HashMap<String, Play> plays = new HashMap<>();
    plays.put("playID", new Play("Play Name", "tragedy"));

    Invoice invoice = new Invoice("Customer Name", List.of(new Performance("playID", 30)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.print(invoice, plays);

    String expectedStatement = "Statement for Customer Name\n" +
            "  Play Name: $400.00 (30 seats)\n" +
            "Amount owed is $400.00\n" +
            "You earned 0 credits\n";

        Assertions.assertEquals(expectedStatement, result);
    }
    
    
    @Test
    void testAudienceForComedy() {
    HashMap<String, Play> plays = new HashMap<>();
    plays.put("playID", new Play("Play Name", "comedy"));

    Invoice invoice = new Invoice("Customer Name", List.of(new Performance("playID", 20)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.print(invoice, plays);

    String expectedStatement = "Statement for Customer Name\n" +
            "  Play Name: $360.00 (20 seats)\n" +
            "Amount owed is $360.00\n" +
            "You earned 4 credits\n"; // Mise à jour de cette ligne en fonction du comportement attendu

        Assertions.assertEquals(expectedStatement, result);
    }
}
