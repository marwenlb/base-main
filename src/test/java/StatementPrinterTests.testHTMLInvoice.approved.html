<html><head><title>Invoice</title></head><body><h1>Invoice</h1><p>Client: BigCo</p><table border="1"><tr><th>Piece</th><th>Seats sold</th><th colspan='2'>Price</th></tr><tr><td>Hamlet</td><td>55</td><td colspan='2'>$650.0</td></tr><tr><td>As You Like It</td><td>35</td><td colspan='2'>$580.0</td></tr><tr><td>Othello</td><td>40</td><td colspan='2'>$500.0</td></tr><tr><td colspan='2'><b>Total Owed</b></td><td >$1730.0</td></tr><tr><td colspan='2'><b>Volume Credits</b></td><td>47</td></tr></table><p>Payment is required under 30 days. We can break your knees if you don't do so.</p></body></html>